src/derivative_operators/BC_operators.jl

```diff
@@ -18,6 +22,7 @@ Creates a periodic boundary condition, where the lower index end of some u is ex
 It is not reccomended to concretize this BC type in to a BandedMatrix, since the vast majority of bands will be all 0s. SpatseMatrix concretization is reccomended.
 """
 struct PeriodicBC{T} <: AtomicBC{T}
```

Comment:
why does it need to know the type? Also, should it should take a dimension?

Reply:
It can take a dimension, it just spits out a `MultiDimBC` full of `PeriodicBCs`, unifying the boundaries so that they all end up of the same type which simplifies things higher up the chain, while still allowing arbitrary mixed boundary conditions.

It needs to know the type since `AtomicBC` has a type and `PeriodicBC <: AtomicBC`, otherwise julia would complain that `T` was undefined.

Reply:
I'm still not sure about this. We probably need an `N` on the atomic BCs.